Source/Core/Common/StringUtil.cpp

```diff
@@ -425,44 +427,69 @@ std::string UTF16ToUTF8(const std::wstring& input)
 
 std::wstring CPToUTF16(u32 code_page, const std::string& input)
 {
-  auto const size = MultiByteToWideChar(code_page, 0, input.data(), (int)input.size(), nullptr, 0);
+  auto const size =
+      MultiByteToWideChar(code_page, 0, input.data(), static_cast<int>(input.size()), nullptr, 0);
 
   std::wstring output;
   output.resize(size);
 
   if (size == 0 ||
-      size != MultiByteToWideChar(code_page, 0, input.data(), (int)input.size(), &output[0],
-                                  (int)output.size()))
+      size != MultiByteToWideChar(code_page, 0, input.data(), static_cast<int>(input.size()),
+                                  &output[0], static_cast<int>(output.size())))
   {
     output.clear();
   }
 
   return output;
 }
 
+std::string UTF16ToCP(u32 code_page, const std::wstring& input)
+{
+  auto const size = WideCharToMultiByte(code_page, 0, input.data(), static_cast<int>(input.size()),
+                                        nullptr, 0, nullptr, false);
+
+  std::string output;
+  output.resize(size);
+
+  if (size == 0 ||
+      size != WideCharToMultiByte(code_page, 0, input.data(), static_cast<int>(input.size()),
+                                  &output[0], static_cast<int>(output.size()), nullptr, false))
+  {
+    ERROR_LOG(COMMON, "WideCharToMultiByte Error in String '%s': %d", input.c_str(),
```

Comment:
You have no guarantee that ```GetLastError()``` will be evaluated first. If ERROR_LOG is a macro, it might do other things that might get evaluated before as well.

I'm not sure that the ```c_str()``` will change ```errno``` but with Microsoft implementation I'm ultra paranoid. The only solution I would accept is assigning the return value of ```GetLastError()``` into a variable, first or something along those lines.